src/test/ui/mpsc_stress.rs

```diff
@@ -37,6 +36,7 @@ impl Barrier {
     fn wait(self) {
         self.shared.fetch_add(1, Ordering::SeqCst);
         while self.shared.load(Ordering::SeqCst) != self.count {
+            thread::yield_now();
```

Comment:
Why is this change needed? I don't think I can approve this particular change without input from someone on the relevant Rust team

Reply:
It's needed for SGX since thread scheduling is not preemptive in SGX and some thread(s) may not be able to increment the shared counter if enough threads get to the loop before every participant in the Barrier has gotten a chance to enter wait and increment the counter. In that scenario a few threads will be spinning forever while others will not get a chance to execute.

I don't imagine this change impacts the correctness of the test. We can also make it conditional for SGX target if you prefer.

Reply:
From https://rust.godbolt.org/z/qqW1oc I think I would be happier if we tossed a `#[cfg(target_env = "sgx")]` on here. Not sure how much this stress test intends to rely on hammering the atomic load as tightly as possible.

This test was added in #55963 so if @stepancheg chimes in that it's fine with a yield_now then keep it.